data/utils/formatting.js

```diff
@@ -34,6 +43,61 @@ textFormatting.parseNumber = function (numberString) {
   return number;
 }
 
+var padZero = function(s, n, str){
+  var diff = Math.max( n-String(s).length+1, 0 );
+  return Array(diff).join(str||'0')+s;
+}
+
+textFormatting.parseCharityNumber = function( ccnumber ) {
+  if (typeof(ccnumber) === 'number'){
+    ccnumber = (ccnumber).toString()
+  }
+  if (typeof(ccnumber) !== 'string') {
+    return null;
+  }
+  ccnumber = ccnumber.replace(/ /g, '').toUpperCase().replace('O', '0');
+  if( ccnumber[0] === 'S' ){
+    // Scottish Charity
+
+    // replace any non digits
+    ccnumber = ccnumber.replace(/\D/g, '');
+
+    // remove any leading zeroes
+    ccnumber = ccnumber.replace(/^0+/g, '');
+
+    // pad to 6 digits with 0
+    ccnumber = padZero(ccnumber, 6, '0');
+
+    // add 'SC' to start
+    ccnumber = 'SC' + ccnumber;
+
+    // stop at 8 chars
+    ccnumber = ccnumber.slice(0,8);
+
+  } else {
+    // England and Wales charity
+
+    // todo: split on '-' separator for subsidiary charities
```

Comment:
You're right, it never happens in the download. I think we were envisaging that this function might be useful for parsing user input, which might include a subsid number in v rare cases.